cloudinit/config/cc_apt_configure.py

```diff
@@ -244,11 +245,18 @@ def apply_apt(cfg, cloud, target):
         )
     # GH: 4344 - stop gpg-agent/dirmgr daemons spawned by gpg key imports.
     # Daemons spawned by cloud-config.service on systemd v253 report (running)
-    subp.subp(
-        ["gpgconf", "--kill", "all"],
+    gpg_process_out, _err = subp.subp(
+        ["ps", "-o", "ppid,pid", "-C", "dirmngr", "-C", "gpg-agent"],
         target=target,
         capture=True,
+        rcs=[0, 1],
```

Comment:
Why is rc 1 needed?

Reply:
@TheRealFalcon if ps doesn't find any matching processes with the `-c gpg-agent -c dirmngr` it'll return 1 sort of like grep. So, in the condition where user-data didn't specify apt/keyids, we wouldn't have these spawned processes and we want to ignore that rc 1 error code. The stdout in the case of no matching process names is
```
   PPID     PID
```
So our  re.findall will match no ppid/pid lines anyway and we NOOP in that case.